tests/ui/type-alias-impl-trait/privacy.rs

```diff
@@ -1,8 +1,10 @@
 #![feature(type_alias_impl_trait)]
+#![feature(type_privacy_lints)]
+#![deny(private_interfaces)]
 
 type Foo = (impl Sized, u8);
 pub fn foo() -> Foo {
-    //~^ ERROR private type alias `Foo` in public interface
+    //~^ ERROR type alias `Foo` is more private than the item `foo`
```

Comment:
We should never see type aliases in this context, they should always be normalized away.
This seems to be a bug affecting `#![feature(type_alias_impl_trait)]` specifically.

Reply:
Ping @oli-obk, you probably know right away which normalization we are missing here.

This error should not be reported because both `Sized` and `u8` are public, and `Foo` should be normalized away in this context.

Reply:
This is intended behaviour. type aliases containing opaque types are supposed to be public if they are used in a public interface.

Reply:
We probably need to discuss this separately because it goes pretty strongly against the general logic of this pass.

Reply:
https://rust-lang.zulipchat.com/#narrow/stream/315482-t-compiler.2Fetc.2Fopaque-types/topic/weak.20type.20aliases.20and.20privacy

Started a thread. We may need to revert a previous PR if this scheme is an issue